demo/src/right-resizable-box.js

```diff
@@ -38,19 +39,20 @@ const useStyles = makeStyles((theme) => ({
             backgroundColor: 'rgba(0, 0, 0, 0)', // The handle is invisible (alpha = 0)
             zIndex: 5,
         },
-    },
-    resizePanelHandleIcon: {
+    }),
+    resizePanelHandleIcon: (theme) => ({
         bottom: '50%',
         right: theme.spacing(-1.75),
         position: 'absolute',
         color: theme.palette.text.disabled,
         transform: 'scale(0.5, 1.5)',
-    },
-}));
+    }),
+};
 
-const RightResizableBox = (props) => {
-    const classes = useStyles();
+// TODO can we avoid to define a component just to add sx support ?
+const ResizableBoxSx = styled(ResizableBox)({});
```

Comment:
Instead of doing this and using mergeSx, the documentation suggest to do it in an other way, did you see it ?
https://mui.com/material-ui/customization/how-to-customize/#dynamic-overrides